### ISSUE TITLE
zlib inflate failed invalid block type

### ISSUE THREAD

**Issue by lmdu** [pauldmccarthy/indexed_gzip#23]
Hi,
Thank you for developing a such nice tool. Recently, I used zran.c and zran.h in my project. I used GCC 8.0.1 in mingw64 to compile code on Windows. It works well with Python3 on Windows. However, it can not work with python2.7. I opened zran_log and recompiled the code. zran_log gives a message:

```
zran_seek(147, 0)
_zran_get_point_with_expand(147, 0, autobuild=1)
_zran_get_point_at(147, c=0)
zran_read(343)
_zran_get_point_with_expand(147, 0, autobuild=1)
_zran_get_point_at(147, c=0)
Discarding 147 bytes (0 < 147)
_zran_inflate(19, block=0, use_offset=1, init_stream=1,
              free_stream=0, init_readbuf=1, free_readbuf=0,
              clear_offsets=1, nbytes=147)
_zran_get_point_at(19, c=1)
initialising to inflate from c=19, u=0
_zran_init_zlib_inflate(19, 19, 0, -15)
Reading from file 19 [ == 19?]
Read 29766 bytes from file [c=19, u=0]
Before inflate - avail_in=29766, avail_out=147
After inflate - avail_in=29765, avail_out=147
zlib inflate failed (code: -3, msg: invalid block type)
```

On Linux, it works well for python2.7. Do you have any suggestions. Thank you again

**Comment by pauldmccarthy** [comment]
Unfortunately I don't have a Windows machine that I can test/debug on, so there's little to no chance that I can solve this on my own. If you could share the `.gz` file which triggered the error, I'll try and reproduce it though.

Also note that Python 2.7 is also being EOLed at the end of 2019 (22 days from now), so you really should consider working with Python 3 exclusively.

**Comment by lmdu** [comment]
Although Python 2.7 is being end of life, it is important for us to performing biological analysis. I have tried many ways to solve this problem, however, nothing changed. I really hope you can help us. Thank you! I have attached the gzip file that triggered the error.
[test.fa.gz](https://github.com/pauldmccarthy/indexed_gzip/files/3964955/test.fa.gz)

**Comment by pauldmccarthy** [comment]
Unable to reproduce the issue on py27/linux (as you have already reported). However, I may have access to a Windows 10 machine in the coming days to test on. What version of Windows are you using?

**Comment by lmdu** [comment]
Thank you for your reply. I'm using Windows 10. You can test the code on Appveyor with python2.7. The zran_read may return a messy code.

**Comment by pauldmccarthy** [comment]
I set up a Python 2.7.17 test environment on Windows 10 using conda, and
unfortunately couldn't reproduce your problem - I can read your test file
without any problem, and all `indexed_gzip` unit tests pass.

While setting up an environment using mingw, I noticed that the latest version
of zlib available through mingw is 1.2.7. This is quite old, and zlib has had
numerous updates and fixes since then, so I'm wondering if your problem is
being caused by an out-dated version of zlib. Would you be able to check which
version of zlib you are using, and upgrade to the latest version?

**Comment by pauldmccarthy** [comment]
Hi @lmdu, thanks for closing the issue - did you identify the cause of this problem?
